### PR TITLE
Fix/smart wallet address mapping fixes

### PR DESCRIPTION
**Pull request by poocart**
Includes:
1. All mappings in all screens/components for Smart Contract wallet address to matching contact in transaction history entry.
2. From now on send flow if receiver has Smart Wallet address then it will be used instead of key based.
3. Additional improvement for checking Pillar Payment Network in asset send flow.
4. Additional check in Pillar Payment Network send flow if receiver address is not in network then throw alert.
5. Removed access keys in the request payload when requesting Smart Wallet addresses for existing contacts as it is not needed and caused issues when returning right addresses.
6. Other minor cleanups or minor fixes.

### CODE REVIEW

File: src/components/ActivityFeed/ActivityFeed.js [hunk 1]
```diff
@@ -261,6 +268,22 @@ class ActivityFeed extends React.Component<Props, State> {
     }
   };
 
+  findMatchingContact = (address) => {
```
Comment (jsidorenko) [inline src/components/ActivityFeed/ActivityFeed.js]: How about extracting this into a separate function?

File: src/components/EventDetails/EventDetails.js [hunk 2]
```diff
@@ -240,10 +256,35 @@ class EventDetails extends React.Component<Props, {}> {
     navigation.navigate(CHAT, { username: contact.username });
   };
 
+  findMatchingContact = (address) => {
+    const {
+      contacts,
+      contactsSmartAddresses = [],
+    } = this.props;
+    const contact = contacts.find(({ id: contactId, ethAddress }) => {
```
Comment (jsidorenko) [inline src/components/EventDetails/EventDetails.js]: DRY

File: src/utils/accounts.js [hunk 3]
```diff
@@ -57,3 +57,22 @@ export function getActiveAccountWalletId(accounts: Accounts = []): string {
...
+      return 'Smart Wallet';
+    case ACCOUNT_TYPES.KEY_BASED:
+      return 'Key Based account';
+    default:
+      return '';
+  }
+}
+
+export function getUserAccounts(accounts: Accounts) {
+  const accountsWithoutActive = accounts.filter(({ isActive }) => !isActive);
```
Comment (jsidorenko) [inline src/utils/accounts.js]:
```suggestion
  const inactiveAccounts = accounts.filter(({ isActive }) => !isActive);
```

File: src/utils/blockchainNetworks.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+    You should have received a copy of the GNU General Public License along
+    with this program; if not, write to the Free Software Foundation, Inc.,
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+import { BLOCKCHAIN_NETWORK_TYPES } from 'constants/blockchainNetworkConstants';
+import type { BlockchainNetwork } from 'models/BlockchainNetwork';
+
+export const isPillarPaymentNetworkActive = (blockchainNetworks: BlockchainNetwork[]) => {
+  const activeBNetwork = blockchainNetworks.find((network) => network.isActive) || { id: '', title: '' };
```
Comment (jsidorenko) [inline src/utils/blockchainNetworks.js]: could be:
```
  const activeBNetwork = blockchainNetworks.find(({ isActive }) => isActive);
  return get(activeBNetworkId, 'id') === BLOCKCHAIN_NETWORK_TYPES.PILLAR_NETWORK;
```

File: src/utils/feedData.js [hunk 5]
```diff
@@ -18,20 +18,43 @@
...
+  accounts: Accounts,
+  eventType: string,
+) {
+  const userAccounts = getUserAccounts(accounts);
   const concatedHistory = history
     .map(({ ...rest }) => ({ ...rest, type: eventType }))
     .map(({ to, from, ...rest }) => {
-      const contact = contacts.find(({ ethAddress }) => {
-        return addressesEqual(from, ethAddress) || addressesEqual(to, ethAddress);
+      const contact = contacts.find(({ id: contactId, ethAddress }) => {
```
Comment (jsidorenko) [inline src/utils/feedData.js]: looks like DRY

File: src/screens/Asset/Asset.js [hunk 6]
```diff
@@ -104,6 +107,7 @@ type Props = {
   history: Array<*>,
   logScreenView: (contentName: string, contentType: string, contentId: string) => void,
   availableStake: number,
+  contactsSmartAddresses: ContactSmartAddresses[],
```
Comment (jsidorenko) [inline src/screens/Asset/Asset.js]: Type should be `ContactSmartAddresses` or `ContactSmartAddress`

File: src/utils/accounts.js [hunk 7]
```diff
@@ -57,3 +58,31 @@ export function getActiveAccountWalletId(accounts: Accounts = []): string {
...
+    case ACCOUNT_TYPES.SMART_WALLET:
+      return 'Smart Wallet';
+    case ACCOUNT_TYPES.KEY_BASED:
+      return 'Key Based account';
+    default:
+      return '';
+  }
+}
+
+export function getInactiveUserAccounts(accounts: Accounts): UserAccount[] {
```
Comment (jsidorenko) [inline src/utils/accounts.js]: This method is so confusing. What is the difference between Account and UserAccount? We need to name the second better

File: src/utils/accounts.js [hunk 8]
```diff
@@ -57,3 +58,31 @@ export function getActiveAccountWalletId(accounts: Accounts = []): string {
...
+  return accounts
+    .filter(({ isActive }) => !isActive)
+    .map(account => ({
+      isUserAccount: true,
+      ethAddress: getAccountAddress(account),
+      username: getAccountName(account.type),
```
Comment (jsidorenko) [inline src/utils/accounts.js]: Username that could be either `Smart Wallet` or `Key Based account`? Should be renamed to `label` or whatever, but not `username`

File: src/utils/blockchainNetworks.js [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+
+    You should have received a copy of the GNU General Public License along
+    with this program; if not, write to the Free Software Foundation, Inc.,
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+import { BLOCKCHAIN_NETWORK_TYPES } from 'constants/blockchainNetworkConstants';
+import type { BlockchainNetwork } from 'models/BlockchainNetwork';
+
+export const isPillarPaymentNetworkActive = (blockchainNetworks: BlockchainNetwork[]): boolean => {
+  return !!blockchainNetworks.find(({ isActive, id }) => !!isActive && id === BLOCKCHAIN_NETWORK_TYPES.PILLAR_NETWORK);
```
Comment (jsidorenko) [inline src/utils/blockchainNetworks.js]:
```suggestion
  return blockchainNetworks.some(({ isActive, id }) => !!isActive && id === BLOCKCHAIN_NETWORK_TYPES.PILLAR_NETWORK);
```

File: src/screens/Asset/Asset.js (hunk 6 above)
Comment (poocart) [reply src/screens/Asset/Asset.js]: After discussions renamed to `ContactSmartAddressData`.

**Pull request merged**